types/src/transaction/mod.rs

```diff
@@ -59,6 +59,25 @@ pub type AtomicVersion = AtomicU64;
 // In StateDB, things readable by the genesis transaction are under this version.
 pub const PRE_GENESIS_VERSION: Version = u64::max_value();
 
+pub trait NextStateVersion {
```

Comment:
Having a trait for this function alone which is implemented by just `transaction` is over-fancy IMO :D